practice/react-app/src/components/Comments.tsx

```diff
@@ -0,0 +1,96 @@
+import axios from 'axios';
+import { useMutation } from '@tanstack/react-query';
+import { useComments } from '../hooks/useComments';
+import { useEffect, useState } from 'react';
+import { Placeholder } from './Placeholder';
+
+const COMMENTS_API = 'http://localhost:3000/api/data/comments';
+
+export const Comments = () => {
+  const [newComment, setNewComment] = useState('');
+  const [newRating, setNewRating] = useState('');
+  const mutation = useMutation({
```

Comment:
opakowałbym to w hooka i wywalił poza komponent tutaj akurat nie używasz tego więcej niż raz, no ale SRP

```
const useAddComment = (setComments, comments) => {
  const queryClient = useQueryClient();

  const mutation = useMutation({
    mutationFn: (comment) => axios.post(COMMENTS_API, comment),
    onSuccess: (data) => {
      queryClient.invalidateQueries('comments');
    },
  });

  return addCommentMutation;
};
```

mając taki hook możesz go używać w kilku miejscach i w zależności od przypadku użycia wrzucać inne callbacki dla onSuccess, onError itd (to tylko przykład)

```
  const addCommentMutation= useAddComment();

  const handleAddComment = (newComment) => {
    addCommentMutation.mutate(newComment, {
      onSuccess: (data) => {
        const addedComment = JSON.parse(data.config.data);
        setComments((prevComments) => [...prevComments, addedComment]);
      },
    });
  };
  
  
```

Reply:
U najs nie wiedziałem że ktoś na to spojrzy, przyznam szczerze z braku czasu nie chciało mi się bardziej tego refactorować, rozbijać na funkcje, robić obsługi błędów itd ale jak znajdę czas chętnie to zrobię